### PR TITLE
[Internal] docs: Add address caches conceptual flow

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Add address caches conceptual flow with code links directly from the nodes/components

### CODE REVIEW

File: docs/SdkDesign.md [hunk 1]
```diff
@@ -17,6 +17,77 @@ flowchart LR
     TransportClient <--> Service[(Cosmos DB Service)]
 ```
 
+## Address caches coneptual model
+
+```mermaid
+flowchart LR
+    subgraph CDB_account
+        subgraph CDB_Account_Endpoint
+            CDB_EP[[fa:fa-cloud CosmosDB-Account/Endpoint]]
```
Comment (kirankumarkolli) [inline docs/SdkDesign.md:26]: GitHUb is not rendering the icons :-( 
Lets remove them.

File: docs/SdkDesign.md [hunk 2]
```diff
@@ -17,6 +17,74 @@ flowchart LR
     TransportClient <--> Service[(Cosmos DB Service)]
 ```
 
+## Address caches coneptual model
```
Comment (ealsur) [inline docs/SdkDesign.md:20]:
```suggestion
## Address caches conceptual model
```

File: docs/SdkDesign.md [hunk 3]
```diff
@@ -17,6 +17,74 @@ flowchart LR
...
+![image](https://user-images.githubusercontent.com/6880899/199167007-bcc054c3-ecb1-4469-ba7d-eae52362e9cd.png)
+
+
+- CollectionCache: Dictionary<CollectionName/Rid, CollectionProperties>
+- CollectionRoutingMap: Single collection PartitionKeyRanges map
+- PartitionKeyRangeCache: Dictionary<CollectionName/Rid, CollectionRoutingMap>
+- GlobalPartitionEndpointManager: Per partition override state. Every reqeust will flow throgh
+    -   Today GlobalEndpointManager is at region scope only and doesn't look at the partition
+    -   Ideal abstraction is to fold it into GlobalEndpointManager --> extra hash computation
+        - Posible to refractor direct code and flow HashedValue down stream (more contract work with direct package)
```
Comment (ealsur) [inline docs/SdkDesign.md:31]:
```suggestion
        - Posible to refactor direct code and flow HashedValue down stream (more contract work with direct package)
```

File: docs/SdkDesign.md [hunk 4]
```diff
@@ -17,6 +17,74 @@ flowchart LR
...
+        GACN[<a href='https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs'>GatewayAddressCache/RN</a>]
+
+        GAC1 --> |NonBlockingAsyncCache| CDBR1_GW
+        GAC1 -.-> CDBR1_BE
+        GACN -->  |NonBlockingAsyncCache| CDBRN_GW
+        GACN -.-> CDBRN_BE
+    end
+```
+
+### Sequence of interaction is
```
Comment (ealsur) [inline docs/SdkDesign.md:62]:
```suggestion
### Sequence of interaction
```

**Review by ealsur — COMMENTED** [review]
Should we just create another md file and reference it from the main diagram? I'm afraid this might keep growing and growing as we add more stuff

**Review by ealsur — APPROVED** [review]

**Pull request merged**